Rover/system.cpp

```diff
@@ -219,6 +219,18 @@ bool Rover::set_mode(Mode &new_mode, ModeReason reason)
     }
 
     Mode &old_mode = *control_mode;
+
+    // HOLD mode in AUTO will keep reversed flag
+    if (!((old_mode.mode_number() == Mode::Number::AUTO && new_mode.mode_number() == Mode::Number::HOLD) ||
+        (old_mode.mode_number() == Mode::Number::HOLD && new_mode.mode_number() == Mode::Number::AUTO))) {
```

Comment:
```suggestion
    if (!((old_mode == &mode_auto && new_mode == &mode_hold) ||
        (&mode_hold== Mode::Number::HOLD && new_mode == &mode_auto))) {
```
Pointer comparison is better.